internal/speakerlist/speakerlist.go

```diff
@@ -80,36 +83,107 @@ func New(logger gokitlog.Logger, nodeName, bindAddr, bindPort, secret, namespace
 }
 
 // Start starts the needed goroutines
-func (sl *SpeakerList) Start(client *k8s.Client) error {
+func (sl *SpeakerList) Start(client *k8s.Client) {
 	sl.client = client
 
 	if sl.mList == nil {
-		return nil
+		return
 	}
 
 	go sl.watchEvents()
+	go sl.memberlistJoinLoop()
+}
 
-	iplist, err := sl.client.GetPodsIPs(sl.namespace, sl.labels)
-	if err != nil {
-		sl.l.Log("op", "startup", "error", err, "msg", "failed to get PodsIPs")
-		return err
+func (sl *SpeakerList) memberlistJoin() {
+	if sl.mList == nil {
+		return
+	}
+	joinList := sl.joinList()
+	if len(joinList) > 0 {
+		n, err := sl.mList.Join(joinList)
```

Comment:
I guess this doesn't block but has a timeout, right? Do you happen to remember what it is? I mean, to have in mind how long it can take to complete

Reply:
10s per Member we are trying to join right now

Reply:
If you have the doc handy and is straightforward, can you add a comment saying it is 10s by default and the link? Does it make sense for you too?

Reply:
Reading that https://github.com/hashicorp/memberlist/issues/105 and the code, not sure it was in the doc

Reply:
done